src/upload-entity-values.js

```
@@ -0,0 +1,84 @@
+//
+// See LICENSE for details
+"use strict";
+
+const Tp = require('thingpedia');
+const fs = require('fs');
+const FormData = require('form-data');
+const csvstringify = require('csv-stringify');
+
+module.exports = {
+    initArgparse(subparsers) {
+        const parser = subparsers.addParser('upload-entity-values', {
+            addHelp: true,
+            description: "Upload a new entity type to Thingpedia."
+        });
+        parser.addArgument('--csv', {
+            required: false,
+            help: "The .csv file of the entity values to upload."
+        });
+        parser.addArgument('--json', {
+            required: false,
+            help: "The .json file of the entity values to upload"
+        });
+        parser.addArgument('--entity-id', {
+            required: true,
+            help: "The id (type) of the string dataset."
+        });
+        parser.addArgument('--entity-name', {
+            required: true,
+            help: "The name of the string dataset."
+        });
+        parser.addArgument('--no-ner-support', {
+            nargs: 0,
+            action: 'storeTrue',
+            help: 'If this entity is an opaque identifier that cannot be used from natural language.',
+            defaultValue: false
+        });
+    },
+
+    async createUpload(args) {
+        const fd = new FormData();
+        if (args.csv) {
+            fd.append('upload', fs.createReadStream(args.csv), {
+                filename: 'entity.csv',
+                contentType: 'text/csv;charset=utf8'
+            });
+        } else if (args.json) {
+            const json = JSON.parse(fs.readFileSync(args.json, 'utf8'));
+            const csv = json.data.map((row) => [row.value, row.name]);
+            const string = csvstringify(csv, {  delimiter:',' });
+            fd.append('upload', string, {
+                filename: 'entity.csv',
+                contentType: 'text/csv;charset=utf8'
+            });
+        }
```

Maybe throw an error if neither --csv not --json is passed?